packages/kbn-eslint-plugin-eslint/rules/no_restricted_paths.js

```diff
@@ -97,11 +97,16 @@ module.exports = {
     }
 
     function checkForRestrictedImportPath(importPath, node) {
-      const absoluteImportPath = resolve(importPath, context);
-      if (!absoluteImportPath) return;
+      let absoluteImportPath;
+      if (importPath.indexOf('ui/') === 0) {
```

Comment:
What's the importance of this line? Should this instead detect if this is a relative or absolute import? Maybe by checking if it starts with a `.`?

Reply:
> What's the importance of this line?

It checks if import is `ui/...`, without this check line 104 returns `undefined`:

```js
absoluteImportPath = resolve(importPath, context);
```

> Should this instead detect if this is a relative or absolute import?

I guess @restrry could comment on that. I guess the idea is to resolve everything to absolute paths and then work with absolute paths. Only the problem I had with that in this PR is that `ui/...` is no resolved, that's why this check. Maybe I can "resolve" `ui/..` to absolute paths too, instead.

Reply:
Actually, absolute paths are made relative back, on line 113, hmmm.

Reply:
I don't know, what is your suggestion, how to expand `ui/kfetch` to `src/legacy/ui/public/kfetch`?

Reply:
paths are:
1. resolved to absolute ones
2. resolved to relative path from `basePath` (kibana folder by default)

Node.js doesn't know about Webpack aliases and you have to manually declare matching 
`ui/*` --> `src/legacy/ui/public/*`

Reply:
Why can't we just say "you can't use imports starting with `ui/*` **and** you can't import relatively from the `src/legacy/ui/public` directory"?

It just feels weird to have config in both `.eslintrc.js` and `packages/kbn-eslint-plugin-eslint/rules/no_restricted_paths.js`

Reply:
> Why can't we just say "you can't use imports starting with `ui/*` and you can't import relatively from the `src/legacy/ui/public` directory"?

That's what this change does. Changes in `packages/*` are necessary to resolve `ui/*` import to something, otherwise it is resolves `absoluteImportPath` to `undefined` and `ui/*` are never processed.

> It just feels weird to have config in both `.eslintrc.js` and `packages/kbn-eslint-plugin-eslint/rules/no_restricted_paths.js`

Config is just in `.eslintrc.js`, in `packages/**` is a fix to not simply *ignore* `ui/*` imports, but resolve them to something.

Reply:
Discussed with @spalger to refactor this file such that it knows nothing about `ui/*`. Implemented.